src/test/java/bmsstream/SenderSeviceTest.java

```diff
@@ -0,0 +1,81 @@
+package bmsstream;
+
+import static bmsstream.data.RandomDataProvider.*;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import org.junit.Test;
+
+import bmsstream.data.DataProviderResolver;
+
+public class SenderSeviceTest 
+{
+    @Test
+    public void whenLineParamZero_expectEmptyStream()
+    {
+    	SenderService.setProvider(DataProviderResolver.resolve("R", "0"));
+        assertTrue( SenderService.stream().size() == 0 );
+    }
+    
+    @Test
+    public void whenLineParamNegative_expectEmptyStream()
+    {
+    	SenderService.setProvider(DataProviderResolver.resolve("R", "-1"));
+        assertTrue( SenderService.stream().size() == 0 );
+    }
+    
+    @Test
+    public void whenLineParamPositive_expectSameStreamSize()
+    {
+    	Integer nline = 10;
+    	SenderService.setProvider(DataProviderResolver.resolve("R", nline.toString()));
+        assertTrue( SenderService.stream().size() == nline );
+    }
+    
+    @Test
+    public void whenRandomProvider_expectValuesWithinRange()
+    {
+    	Integer nline = 10;
+    	SenderService.setProvider(DataProviderResolver.resolve("R", nline.toString()));
+        List<String> result = SenderService.stream();
+        for(String line : result)
+        {
+        	String[] params = line.split(DELIMITER);
+        	Integer temp = Integer.parseInt(params[0]);
+        	Integer soc = Integer.parseInt(params[1]);
+        	Float cr = Float.parseFloat(params[2]);
+        	assertTrue(checkInRange(MIN_TEMP,temp,MAX_TEMP));
```

Comment:
These tests are checking random number generation. They may pass even when the ranges are wrong. Focus on testing your functionality - the most crucial is the output format.
In the absence of a test for the output format, any future mistakes will be caught only while integrating with the receiver.